configs/phpcs.xml

```diff
@@ -2,23 +2,18 @@
 <ruleset>
   <description>PHP CodeSniffer ruleset for Drupal.</description>
 
-  <rule ref="vendor/drupal/coder/coder_sniffer/Drupal"  />
+  <!-- Exclude unsupported file types. -->
+  <exclude-pattern>*.gif</exclude-pattern>
+  <exclude-pattern>*.less</exclude-pattern>
+  <exclude-pattern>*.png</exclude-pattern>
 
-  <rule ref="vendor/drupal/coder/coder_sniffer/DrupalPractice">
-    <exclude name="DrupalPractice.Yaml.RoutingAccess.OpenCallback" />
-    <exclude name="DrupalPractice.FunctionDefinitions.AccessHookMenu" />
-  </rule>
+  <!-- Minified files don't have to comply with coding standards. -->
+  <exclude-pattern>*.min.css</exclude-pattern>
+  <exclude-pattern>*.min.js</exclude-pattern>
 
-  <rule ref="Drupal.Files.LineLength">
-    <properties>
-      <property name="lineLimit" value="120" />
-      <property name="absoluteLineLimit" value="150" />
-    </properties>
-  </rule>
+  <rule ref="./vendor/drupal/coder/coder_sniffer/Drupal" />
+  <rule ref="./vendor/drupal/coder/coder_sniffer/DrupalPractice"></rule>
 
-  <rule ref="Drupal.Arrays.Array">
-    <properties>
-      <property name="lineLimit" value="120" />
-    </properties>
-  </rule>
+  <!-- PSR-12 sniffs -->
+  <rule ref="PSR12.Files.DeclareStatement" />
```

Comment:
@zero2one, als we dit hier toevoegen gaan we behoorlijk wat werk veroorzaken op alle bestaande projecten.
Is er een tool die dit automatisch kan op basis van de phpdoc, phpcbf ofzo?

Reply:
This does not force you to put the declare statement in files, it only checks if it is written correctly when it the declare statement is in the file.